File: backend/Entities/Patient.cs
namespace ClinicFlow.Api.Entities;

/// <summary>
/// A clinic patient record. Medical fields are plain free text only — this
/// app does not perform diagnosis or store structured clinical data.
/// </summary>
public class Patient
{
    public Guid Id { get; set; }
    public required string FirstName { get; set; }
    public required string LastName { get; set; }
    public required string PhoneNumber { get; set; }
    public string? Email { get; set; }
    public PatientGender Gender { get; set; }
    public DateOnly? DateOfBirth { get; set; }
    public string? Address { get; set; }
    public string? EmergencyContactName { get; set; }
    public string? EmergencyContactPhone { get; set; }
    public string? MedicalNotes { get; set; }
    public string? Allergies { get; set; }
    public bool IsActive { get; set; } = true;
    public DateTime CreatedAtUtc { get; set; }
    public DateTime? UpdatedAtUtc { get; set; }
}

File: frontend/src/types/patient.ts
export type PatientGender = "Male" | "Female" | "Other" | "PreferNotToSay";

export interface Patient {
  id: string;
  firstName: string;
  lastName: string;
  fullName: string;
  phoneNumber: string;
  email: string | null;
  gender: PatientGender;
  dateOfBirth: string | null;
  address: string | null;
  emergencyContactName: string | null;
  emergencyContactPhone: string | null;
  medicalNotes: string | null;
  allergies: string | null;
  isActive: boolean;
  createdAtUtc: string;
  updatedAtUtc: string | null;
}

export interface PatientListItem {
  id: string;
  fullName: string;
  phoneNumber: string;
  email: string | null;
  gender: PatientGender;
  dateOfBirth: string | null;
  isActive: boolean;
  createdAtUtc: string;
}

export interface CreatePatientRequest {
  firstName: string;
  lastName: string;
  phoneNumber: string;
  email?: string;
  gender: PatientGender;
  dateOfBirth?: string;
  address?: string;
  emergencyContactName?: string;
  emergencyContactPhone?: string;
  medicalNotes?: string;
  allergies?: string;
}

export interface UpdatePatientRequest extends CreatePatientRequest {}

export interface PatientQuery {
  search?: string;
  isActive?: boolean;
  gender?: PatientGender;
  pageNumber?: number;
  pageSize?: number;
}

export interface PatientStats {
  totalPatients: number;
  activePatients: number;
  inactivePatients: number;
  newPatientsThisMonth: number;
}

export interface PaginatedResponse<T> {
  items: T[];
  pageNumber: number;
  pageSize: number;
  totalCount: number;
  totalPages: number;
}
